File: Java/프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        Deque<Integer> queue = new LinkedList<>();
        for (int temp : priorities) {
            queue.addLast(temp);
        }
        
        PriorityQueue<Integer> priorityQueue = new PriorityQueue<>(Collections.reverseOrder());
        for (int priority : priorities) {
            priorityQueue.add(priority);
        }
        
        int answer = 0;
        while (!queue.isEmpty()) {
            int current = queue.pollFirst();
            if (current == priorityQueue.peek()) {
                priorityQueue.poll();
                answer++;
                if (location == 0) {
                    break;
                }
            } else {
                queue.addLast(current);
            }
            location = (location - 1 + queue.size()) % queue.size(); // location 업데이트
        }
        return answer;
    }
}

File: Python3/프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    priority = list(priorities)
    pq = deque() # save a list [문자 , 우선순위] 
    for x in range(len(priorities)):
        pq.append([x, priorities[x]])

    answer = 0
    priority.sort()
    
    while(pq):
        if pq[0][1] != priority[-1]:
            pq.append(pq.popleft())
        else:
            temp = pq.popleft()
            priority.pop()
            if temp[0] == location:
                print(priority)
                print(pq)
                answer+=1
                break
            else:
                answer += 1
    return answer